lts.js

```diff
@@ -0,0 +1,79 @@
+const fs = require("fs/promises");
+const path = require("path");
+
+(async () => {
+  const LTS_SCHEDULE_URL =
+    "https://raw.githubusercontent.com/nodejs/Release/main/schedule.json";
+
+  const versions = await fetch(LTS_SCHEDULE_URL).then((r) => r.json());
+
+  const currentLTS = +Object.entries(versions)
```

Comment:
The pipeline operator will eventually make it so much easier to write this kind of code cleanly without creating a gazillion temporary variables. But until then... you either get these chains or you have intermediate variables. But you've given me some good inspiration for how to restructure this to be easier to follow (hopefully).

Reply:
Re: testing, testing a test feels weird? But now I'm curious if there are standard practices for testing CI/CD code.

Reply:
well, in my thinking, this code isn't explicitly a test or part of CI/CD -- it's just javascript code to check versions, even though it's being utilized in a workflow. in general, when i've written any kinda infrastructure code, if i can pull out methods/functions to test them, that's ideal.

it just feels like there's a lot of code in here that feels like it could go wrong if things aren't structured exactly as they are currently envisioned. it could also make it clearer what was expected as inputs and outputs for someone onboarding onto this code! 

i also bet there are some ways of doing things like ... testing if a particular workflow ran, although i don't know if that's exactly what you're asking. (there is also definitely stuff like [serverspec](https://serverspec.org/), which is explicitly writing tests against your infrastructure, to test things like your deployment code.)

Reply:
I guess I'm struggling with when do you stop testing, then? A kind of "who tests the tests?" thing. This code isn't part of the product and it isn't creating infrastructure. It's checking that the things that do create infrastructure are doing so with the expected Node.js version. If this code fails, it doesn't create new harm, it just fails to add new value.

I'm not opposed to putting in tests, to be clear. It's just a murky boundary. Would it feel cleaner if we moved this JS file into the `.github/workflows` directory?

Reply:
i guess i'm thinking less about harm and more about debugging and maintainability/clarity of code. if in the future this code needs to be changed, that would feel difficult and dangerous to me, so i wouldn't want to do it. if there were tests that i could update to make sure the outputs/inputs were what i wanted them to be, i would feel more comfortable doing it. but it's fine if you don't want to add them! it was just a suggestion.